Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/TapToPlace/TapToPlace.cs

```diff
@@ -0,0 +1,467 @@
+        }
+
+        private void StartPlacement()
+        {
+            CoreServices.InputSystem?.RegisterHandler<IMixedRealityPointerHandler>(this);
+
+            // Temporarily change the game object layer to IgnoreRaycastLayer to enable a surface hit beyond the game object
+            gameObject.layer = IgnoreRaycastLayer;
+
+            if (SpatialMeshVisible)
+            {
+                ToggleSpatialMeshVisibility(true);
+            }
+
+            SolverHandler.UpdateSolvers = true;
+
+            IsBeingPlaced = true;
+        }
+
+        private void StopPlacement()
+        {
+            // Change the game object layer back to default which enables raycast hits again
+            gameObject.layer = DefaultLayer;
+
+            if (SpatialMeshVisible)
+            {
+                ToggleSpatialMeshVisibility(false);
+            }
+
+            SolverHandler.UpdateSolvers = false;
+
+            IsBeingPlaced = false;
+
+            CoreServices.InputSystem?.UnregisterHandler<IMixedRealityPointerHandler>(this);
+        }
+
+        /// <inheritdoc/>
+        public override void SolverUpdate()
+        {
+            // Make sure the Transform target is not null, added for the case where auto start is true 
+            // and the tracked target type is the controller ray, if the hand is not in the frame we cannot
+            // calculate the postion of the object
+            if (SolverHandler.TransformTarget != null)
+            {
+                PerformRaycast();
+                SetPosition();
+                SetRotation();
+            }
+        }
+
+        protected virtual void PerformRaycast()
+        {
+            // The transform target is the transform of the TrackedTargetType, i.e. Controller Ray, Head or Hand Joint
+            var transform = SolverHandler.TransformTarget;
+
+            Vector3 origin = transform.position;
+            Vector3 endpoint = transform.position + transform.forward;
+            currentRay.UpdateRayStep(ref origin, ref endpoint);
+
+            // Check if the current ray hits a magnetic surface
+            DidHit = MixedRealityRaycaster.RaycastSimplePhysicsStep(currentRay, MaxRaycastDistance, MagneticSurfaces, false, out currentHit);  
+        }
+
+        protected virtual void SetPosition()
+        {
+            // Change the position of the GameObjectToPlace if there was a hit, if not then place the object at the default distance
+            // relative to the TrackedTargetType position
+            
+            if (DidHit)
+            {
+                // take the current hit point and add an offset relative to the surface to avoid half of the object in the surface
+                GoalPosition = currentHit.point;  
+                AddOffset(currentHit.normal * SurfaceNormalOffset);
+                
+                // Draw the normal of the raycast hit for debugging 
+                Debug.DrawRay(currentHit.point, currentHit.normal * 0.5f, Color.yellow);
```

Comment:
we might want to wrap these with #IF UNITY_EDITOR or check they are compiled out at runtime

Also maybe add DebugEnabled property to toggle

Reply:
DebugEnabled property added